src/main/java/com/backend_training/app/services/PostService.java

```diff
@@ -0,0 +1,70 @@
+package com.backend_training.app.services;
+
+import com.backend_training.app.dto.PostResponse;
+import com.backend_training.app.models.Post;
+import com.backend_training.app.repositories.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class PostService {
+
+    @Autowired
+    private PostRepository postRepository;
+
+
+    public Post createPost(Post post) {
+        postRepository.save(post);
+        return post;
+    }
+
+    public Post updatePost(Post post) throws Exception {
+        if (postRepository.existsById(String.valueOf(post.getId()))) {
+            postRepository.save(post);
+            return post;
+        } else {
+            throw new Exception("Post not found");
+        }
+    }
+
+    public Post deletePost(UUID id) throws Exception {
+        Post post = postRepository.findById(id);
+        if (post != null) {
+            post.setDeleted(true);
+            postRepository.save(post);
+            return post;
+        } else {
+            throw new Exception("Post not found");
+        }
+    }
+
+    public Post getPost(UUID id) throws Exception {
```

Comment:
```suggestion
 public Post getPost(UUID id) throws Exception {
        Post post =  postRepository.findById(id);
        if (post == null || post.isDeleted()) {
              throw new Exception("Post not found");
        } 
        
        return post;
    }
```
For better readability, even in other places in update, delete and create method as well.

Reply:
Thanks for the suggestion!
I've implemented the changes here:
https://github.com/George1010/PostIt/pull/4/files#diff-c0dcadec33746c3b27a2badaea2aad51ddac36528acd487620d770e087f84807